### ISSUE TITLE
Pace is not using the dedicated field but comes from GPS ?

### ISSUE THREAD

**Issue by risbac** [jimmykane/quantified-self#200]
Hello !

https://quantified-self.io/user/3Q1MQ5elv5U33jgMDJMe41jb8r52/event/PsFSJBibSAUKLCbutIxy

In this run, I'm not moving at all, between minutes 23 and 24. I have a 0 speed in the fit file, but in your chart, I see a pace. The value is calculated from the GPS data maybe ? Would it be possible to use the pace field when it's calculated ? I have a Stryd, and I can't compare how good is the instant pace of other watches, since the value from the watch with the Stryd is not the right one. Or am I missing a way to use that data in the chart ? 

Thanks in advance for any feedback !

**Comment by jimmykane** [comment]
Hey could you make the link public so I can check it out ?

If this is a fit file then no calculation should happen. It should in
theory show what the watch / stryd recorded.

**Comment by risbac** [comment]
Oops sorry, I thought that I made it public. That's done now !
It's indeed a fit file. I'm attaching it here as well.

**Comment by jimmykane** [comment]
Ok I took a little look. Its ok.

What you see there is that there is speed/pace missing. So the chart just
joins with a straight line the 2 points.

If you would hover your mouse over those minutes you would see no
pace/speed statistics.

The thing is that you had a speed of 0 if not moving. That means a pace of
"Infinity". Since infinity is not able to be charted it is just left as it
is, when pace / speed is missing (lets assume they are 0).

Does this make a little more sense?

I could improve this though

**Comment by risbac** [comment]
Ah yes indeed, it makes sense ! Pace is a problem, there is no way to show
infinity. In some applications I have used, they cheat, and show a value,
but a big one (like 1h/km). But often it messes with the scale and it's
hard to see the variations of pace. I think that it would be better to show
a "hole" in the chart. Right now it's a bit misleading, don't you think?

**Comment by jimmykane** [comment]
Yeah it needs a break on the chart but but but I was planning to implement
to show pauses , a gray area. I suppose that you didn't pause there. So
that would not help.

The thing for putting a gap there is that it might just create some issues
on other places the pace is missing. Eg a smaller "pause" , a hard uphill
etc. I'll do some tests and keep this in mind.

Let's leave this issue open.

What do you think?

**Comment by risbac** [comment]
>Yeah it needs a break on the chart but but but I was planning to implement
to show pauses , a gray area. I suppose that you didn't pause there. So
that would not help.

Well... I was not moving. So it's a pause in a way. But since I run on
trails mostly, when I stop running, I don't pause the watch, it's a part of
the recovery, so I don't see why I would exclude that from the calculation.
If people are pausing the watch, that's different I guess, it means
"exclude the data during the pause". In my case, I'm fine with the average
values including my pause. But it's just a bit misleading to see a pace
value on the chart when actually there is none and it makes it difficult to
compare 2 devices to see if they react to the instant pace the same way.
Maybe you can put it in "light grey", but while still using the values to
calculate the average values. And in the case of a real pause, you show in
grey, but you exclude them from the calculation. A bit more complicated
technically speaking, but maybe that would be the best compromise.

**Comment by jimmykane** [comment]
I understand well and I agree about recovery, vs pausing.

I'll see what I can do.

Regarding the calculations, when the pace is missing it should not affect
the selectable chart area nor the generic stats.
